utils/swift_build_support/swift_build_support/build_script_invocation.py

```diff
@@ -766,10 +771,13 @@ def _execute_merged_host_lipo_core_action(self):
         self._execute_action("merged-hosts-lipo-core")
 
     def _execute_action(self, action_name):
+        log_time('start', action_name)
```

Comment:
1. Can you use a with statement to make this scope based rather than inlining this all the time? Something like this: https://docs.python.org/3/library/contextlib.html:

```
import contextlib
from contextlib import contextmanager

@contextlib.contextmanager
def log_time_in_scope(action_name):
    log_time('start', action_name)
    t_start = time.time()
    try:
        yield
    finally:
        log_time('end', action_name, time.time() - t_start)
```

I would define this as the outside API instead of log_time.

2. Can you add a build system unit test that checks that we emit these correctly? (I am assuming these work even if we dry-run).

Can you fix that and then LGTM.

Reply:
Good point! I will update the PR.